src/en/mangamo/src/eu/kanade/tachiyomi/extension/en/mangamo/Mangamo.kt

```diff
@@ -0,0 +1,373 @@
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+class Mangamo : ConfigurableSource, HttpSource() {
+
+    override val name = "Mangamo"
+
+    override val lang = "en"
+
+    override val baseUrl = "https://mangamo.com"
+
+    override val supportsLatest = true
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    private val helper = MangamoHelper(headers)
+
+    private var userToken = ""
+        get() {
+            if (field == "") {
+                field = preferences.getString(MangamoConstants.USER_TOKEN_PREF, "")!!
+
+                if (field == "") {
+                    field = MangamoAuth.createAnonymousUserToken(client)
+                    preferences.edit()
+                        .putString(MangamoConstants.USER_TOKEN_PREF, field)
+                        .apply()
+                }
+            }
+            return field
+        }
+
+    private val auth by cachedBy({ userToken }) {
+        MangamoAuth(helper, client, userToken)
+    }
+
+    private val firestore by cachedBy({ auth }) {
+        FirestoreRequestFactory(helper, auth)
+    }
+
+    private val coinMangaPref
+        get() = preferences.getStringSet(MangamoConstants.HIDE_COIN_MANGA_PREF, setOf())!!
+    private val exclusivesOnlyPref
+        get() = preferences.getStringSet(MangamoConstants.EXCLUSIVES_ONLY_PREF, setOf())!!
+
+    override val client: OkHttpClient = super.client.newBuilder()
+        .addNetworkInterceptor {
+            val response = it.proceed(it.request())
+
+            // Add Cache-Control to Firestore queries
+            if (it.request().url.toString().startsWith(MangamoConstants.FIRESTORE_API_BASE_PATH)) {
+                return@addNetworkInterceptor response.newBuilder()
+                    .header("Cache-Control", "public, max-age=${MangamoConstants.FIRESTORE_CACHE_LENGTH}")
+                    .build()
+            }
+            response
+        }
+        .build()
+
+    private fun processSeries(dto: SeriesDto) = SManga.create().apply {
+        author = dto.authors?.joinToString { it.name }
+        description = dto.description
+        genre = dto.genres?.joinToString { it.name }
+        status = helper.getSeriesStatus(dto)
+        thumbnail_url = dto.titleArt
+        title = dto.name!!
+        url = helper.getSeriesUrl(dto)
```

Comment:
add `initialized = true` to avoide an extra call for details.

since it appears all the information is already available during browsing